src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al

```diff
@@ -0,0 +1,509 @@
+codeunit 324 "No. Series Copilot Impl."
+{
+    var
+        IncorrectCompletionErr: Label 'Incorrect completion. The property %1 is empty';
+        IncorrectCompletionNumberOfGeneratedNoSeriesErr: Label 'Incorrect completion. The number of generated number series is incorrect. Expected %1, but got %2';
+        TextLengthIsOverMaxLimitErr: Label 'The property %1 exceeds the maximum length of %2';
+        DateSpecificPlaceholderLbl: label '{current_date}', Locked = true;
+        NotAbleToGenerateNumberSeriesTryToRephraseErr: Label 'Sorry, I am not able to generate the number series. Try to rephrase your request or provide more details.';
+
+    internal procedure GetNoSeriesSuggestions()
+    var
+        FeatureTelemetry: Codeunit "Feature Telemetry";
+        NoSeriesCopilotRegister: Codeunit "No. Series Copilot Register";
+        NoSeriesCopilotImpl: Codeunit "No. Series Copilot Impl.";
+        AzureOpenAI: Codeunit "Azure OpenAI";
+        NoSeriesCopilot: Page "No. Series Proposal";
+    begin
+        NoSeriesCopilotRegister.RegisterCapability();
+        if not AzureOpenAI.IsEnabled(Enum::"Copilot Capability"::"No. Series Copilot") then
+            exit;
+
+        FeatureTelemetry.LogUptake('0000LF4', NoSeriesCopilotImpl.FeatureName(), Enum::"Feature Uptake Status"::Discovered); //TODO: Update signal id
+
+        NoSeriesCopilot.LookupMode := true;
+        NoSeriesCopilot.Run();
+    end;
+
+    procedure Generate(var NoSeriesProposal: Record "No. Series Proposal"; var ResponseText: text; var NoSeriesGenerated: Record "No. Series Proposal Line"; InputText: Text)
+    var
+        SystemPromptTxt: SecretText;
+        CompletePromptTokenCount: Integer;
+        Completion: Text;
+        TokenCountImpl: Codeunit "AOAI Token";
```

Comment:
You should order all of your variables properly, or CodeCop will be on your neck ;-)